cylc/flow/platforms.py

```diff
@@ -190,14 +198,14 @@ def platform_from_name(
     if platform_name is None:
         platform_name = 'localhost'
 
-    platform_group = None
     for platform_name_re in reversed(list(platform_groups)):
         # Platform is member of a group.
         if re.fullmatch(platform_name_re, platform_name):
             platform_name = get_platform_from_group(
                 platform_groups[platform_name_re], group_name=platform_name,
                 bad_hosts=bad_hosts
             )
+            break
```

Comment:
I think this will change the order in which platforms match.

Please check whether this matches the order specified in the platforms proposal. If this doesn't break tests we could do with adding one. Might be easier to push to another PR if unrelated.

Reply:
See below https://github.com/cylc/cylc-flow/blob/dcd1290bdefa1a7e00a4c2998b1c22d54b0983c7/cylc/flow/platforms.py#L213-L216

If we are happy the same principle should be applied to platform groups (in which case it was wrong all this time) then a similar comment should be added above this for loop.

Reply:
I'm pretty confident that the same principle should be applied to platform groups, and for the same reason: We want user created platform groups to take precedence over site-admin set ones.

Reply:
```suggestion
```